tests/unit/lms/services/grading_info_test.py

```diff
@@ -100,17 +100,26 @@ def test_it_creates_new_record_if_no_matching_exists(
     def test_it_updates_existing_record_if_matching_exists(
         self, svc, pyramid_request, h_user, lti_user, db_session,
     ):
-        # Update a couple of attributes on the model...
-        h_user = h_user._replace(display_name="Someone Else")
-
-        pyramid_request.POST["lis_result_sourcedid"] = "something different"
+        grading_info = GradingInfo(
+            lis_result_sourcedid="lis_result_sourcedid",
+            lis_outcome_service_url="lis_outcomes_service_url",
+            oauth_consumer_key=lti_user.oauth_consumer_key,
+            user_id=lti_user.user_id,
+            context_id=pyramid_request.params["context_id"],
+            resource_link_id=pyramid_request.params["resource_link_id"],
+            tool_consumer_info_product_family_code="tool_consumer_info_product_family_code",
+            h_username="h_username",
+            h_display_name="h_display_name",
+        )
```

Comment:
This test is hard to grok as it's not clear which arguments are relevant. Using a fixture or method to create this would give an opportunity to give it an helpful name. If you used a method, it's also an opportunity for that method to only accept the parts which are relevant for this test.

My guess is it's only the user name and display name?

Reply:
I think a `GradingInfo` test factory would simplify and clean up these tests quite a bit, but we'd need the [test factories PR](https://github.com/hypothesis/lms/pull/1642) merged first

Reply:
Done (adding a `GradingInfo` factory)

Reply:
Ending up moving the `GradingInfo` factory into a separate PR, because it turns out to be a slightly larger thing: https://github.com/hypothesis/lms/pull/1650

To get this PR merged I just added a quickie helper method to remove noise from this one test: https://github.com/hypothesis/lms/pull/1640/commits/11409dc7335c6e2edb72d59cf02fda8cd369b426

Note there are already other `GradingInfo` factory methods with different features elsewhere in this same test module! Trying to merge them into a single helper method with all the features would be too much work and comes too close to reimplementing the Factory Boy-based `GradingInfo` factory in #1650 with my own code. I'll just get #1650 finished up and merged soon, instead

Reply:
Yeah no worries. Looks like I guessed the wrong fields incidentally, so there we go.